manifests/server.pp

```diff
@@ -63,37 +63,35 @@
     require        => Package['rsync']
   }
 
-  if ($facts['os']['name'] in ['RedHat', 'CentOS']) and (versioncmp($facts['os']['release']['major'], '7') < 0) {
-    file { '/etc/init.d/rsyncd':
-      ensure  => 'file',
-      owner   => 'root',
-      group   => 'root',
-      mode    => '0750',
-      content => file("${module_name}/rsync.init")
-    }
-
+  if 'systemd' in $facts['init_systems'] {
```

Comment:
I flipped the content around, and made systemd the special case, not the default case.